public/app.js

```diff
@@ -0,0 +1,380 @@
+          case 'Songs': return data.items;
+          default: return data;
+
+        }
+    } catch (error) {
+      alert(error.message);
+    }
+    return null;
+  }
+  /**
+   * @description Функция получения Токена Spotify
+   */
+  const _getToken = async () => {
+
+    const result = await fetch('https://accounts.spotify.com/api/token', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+        'Authorization': 'Basic ' + btoa(clientId + ':' + clientSecret)
+      },
+      body: 'grant_type=client_credentials'
+    }).catch(err => alert(err.message));
+
+    const data = await result.json();
+
+    if (data.error) {
+       alert("We cant find your token :(");
+      return null;
+    } else {
+      return data.access_token;
+      }
+
+  }
+  /**
+   * @description Функция получения жанров
+   * @param {string} token - токен Spotify
+   */
+  const _getGenres = async (token) => {
+    return send(`https://api.spotify.com/v1/browse/categories?locale=sv_RU`, token, 'Genre');
+  }
+  /**
+   * @description Функция получения плейлистов по подходящему жанру
+   * @param {string} token - токен Spotify
+   * @param {string} genreId - id заданного жанра
+   */
+  const _getPlaylistByGenre = async (token, genreId) => {
+    return send(`https://api.spotify.com/v1/browse/categories/${genreId}/playlists?limit=10`, token, 'GenrePlaylists');
+  }
+
+  /**
+   * @description Функция получения списка песен
+   * @param {string} token - токен Spotify
+   * @param {string} tracksEndPoint - ссылка для получения API ответа с информацией про песни в плейлисте
+   */
+  const _getTracks = async (token, tracksEndPoint) => {
+
+    return send(`${tracksEndPoint}?limit=10`, token,'Songs');
+  }
+  /**
+   * @description Функция получения информации о песне
+   * @param {string} token - токен Spotify
+   * @param {string} tracksEndPoint - ссылка для получения API ответа с информацией про песню
+   */
+  const _getTrack = async (token, trackEndPoint) => {
+    return send(`${trackEndPoint}`, token);
+  }
+  /**
+   * @description Функция получения информации о плейлисте
+   * @param {string} token - токен Spotify
+   * @param {string} id - id плейлиста
+   */
+  const _getPlaylist = async (token, id) => {
+    return send(`https://api.spotify.com/v1/albums/${id}`, token);
+  }
+
+  return {
+    getToken() {
+      return _getToken();
+    },
+    getGenres(token) {
+      return _getGenres(token);
+    },
+    getPlaylistByGenre(token, genreId) {
+      return _getPlaylistByGenre(token, genreId);
+    },
+    getTracks(token, tracksEndPoint) {
+      return _getTracks(token, tracksEndPoint);
+    },
+    getTrack(token, trackEndPoint) {
+      return _getTrack(token, trackEndPoint);
+    },
+    getPlaylist(token, id) {
+      return _getPlaylist(token, id);
+    }
+  }
+})();
+
+
+const UIController = (function() {
+  const createNewElement = (type, name, img, artist) => {
+
+    const element =
```

Comment:
от этой переменной нет смысла.

Можно сразу возвращать результат
```suggestion
    return 
```